deltacat/tests/aws/test_s3u.py

```diff
@@ -87,6 +97,51 @@ def test_upload_throttled(self, mock_s3_client_cache):
 
         assert mock_s3.put_object.call_count > 3
 
+    @patch("deltacat.aws.s3u.UPLOAD_SLICED_TABLE_RETRY_STOP_AFTER_DELAY", 1)
+    @patch("deltacat.aws.s3u.ManifestEntry")
+    @patch("deltacat.aws.s3u._get_metadata")
+    @patch("deltacat.aws.s3u.CapturedBlockWritePaths")
+    def test_upload_sliced_table_retry(
+        self,
+        mock_captured_block_write_paths,
+        mock_get_metadata,
+        mock_manifest_entry,
+    ):
+        mock_manifest_entry.from_s3_obj_url.side_effect = OSError(
+            "Please reduce your request rate.."
+        )
+        mock_get_metadata.return_value = [mock.MagicMock()]
+        cbwp = CapturedBlockWritePaths()
+        cbwp._write_paths = ["s3_write_path"]
+        cbwp._block_refs = [mock.MagicMock()]
+        mock_captured_block_write_paths.return_value = cbwp
+        with pytest.raises(RetryError):
```

Comment:
I always assumed it would throw the underlying exception when retries are exhausted. Should we then add `RetryError` to ray task level retries here? https://github.com/ray-project/deltacat/blob/main/deltacat/compute/compactor_v2/utils/task_options.py

Reply:
Good idea, added.